api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -118,4 +118,16 @@ public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
    * @return this for method chaining
    */
   ExpireSnapshots cleanExpiredFiles(boolean clean);
+
+  /**
+   * Allows expiration of unreachable metadata, such as partition specs as part of the operation.
+   *
+   * @param clean setting this to true will remove metadata(such as partition spec) that are no
+   *     longer reachable by any snapshot
+   * @return this for method chaining
+   */
+  default ExpireSnapshots cleanExpiredMeta(boolean clean) {
```

Comment:
I'd prefer to use the full `metadata` word, especially in a public API, we don't really get much from an abbreviation.
`cleanExpiredMetadata`

Reply:
+1 to use `cleanExpiredMetadata`